Describe structs

siman opened the pull request.

siman committed:
Add Cargo files

siman committed:
Add .gitignore file

siman committed:
Add LICENSE file

siman committed:
Describe structs of versioned store

bedeho commented on src/lib.rs:
```diff
@@ -0,0 +1,237 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
```
Change copyright to Jsgenesis.

bedeho commented on src/lib.rs:
```diff
@@ -0,0 +1,237 @@
+    // External(ExternalProperty),
+}
+
+impl Default for PropertyType {
```
Why is this needed?

siman replied:
Why do we need a `Default` for `PropertyType`?
Because `PropertType` is an enum and if we want enum to be used in Substrate RPC that it should be serializable. This means that Substrate needs a default value for enums.

bedeho commented on src/lib.rs:
```diff
@@ -0,0 +1,237 @@
+    // External(ExternalPropertyType),
+}
+
+impl Default for PropertyValue {
```
Why is this needed?

siman replied:
Answered above to a similar question.

bedeho reviewed (changes requested):
1. There are 39 warnings, could you try to reduce or eliminate those?

2. Untrack Cargo.lock, libraries should not track it, therefore also add to .gitignore
https://github.com/rust-lang/cargo/issues/315

mnaamani committed:
update to depend on substrate branch v1.0

siman committed:
Merge pull request #1 from mnaamani/update-substrate-dependency

update to depend on substrate branch v1.0

siman committed:
Delete Cargo.lock from repo

siman committed:
Add Cargo.lock to .gitignore

siman committed:
Change copyright header from Parity to Jsgenesis

siman commented:
> 1. There are 39 warnings, could you try to reduce or eliminate those?
> 2. Untrack Cargo.lock, libraries should not track it, therefore also add to .gitignore
>    [rust-lang/cargo#315](https://github.com/rust-lang/cargo/issues/315)

1. Warnings will be fixed in [another PR](https://github.com/Joystream/substrate-versioned-store/pull/1), because there I implement functions and warnings of unused function arguments just disappear. 
2. Fixed: Removed Cargo.lock from repo. Added Cargo.lock to .gitignore.
